server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4717,8 +4717,8 @@ public VirtualMachine migrateVirtualMachineWithVolume(Long vmId, Host destinatio
         }
 
         // Check if the source and destination hosts are of the same type and support storage motion.
-        if (!(srcHost.getHypervisorType().equals(destinationHost.getHypervisorType()) && srcHost.getHypervisorVersion().equals(destinationHost.getHypervisorVersion()))) {
-            throw new CloudRuntimeException("The source and destination hosts are not of the same type and version. " + "Source hypervisor type and version: "
+        if (!(srcHost.getHypervisorType().equals(destinationHost.getHypervisorType()))) {
+            throw new CloudRuntimeException("The source and destination hosts are not of the same type. " + "Source hypervisor type and version: "
```

Comment:
The Xenserver version on the target host can be same or higher, please see if the version check can be further improved.